pkg/cmd/issue/transfer/transfer.go

```diff
@@ -0,0 +1,168 @@
+package transfer
+
+import (
+	"fmt"
+	"net/http"
+	"sort"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/config"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmd/issue/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/pkg/prompt"
+	"github.com/cli/cli/pkg/text"
+	"github.com/spf13/cobra"
+)
+
+type TransferOptions struct {
+	HttpClient func() (*http.Client, error)
+	Config     func() (config.Config, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	IssueSelector    string
+	DestRepoSelector string
+}
+
+func NewCmdTransfer(f *cmdutil.Factory, runF func(*TransferOptions) error) *cobra.Command {
+	opts := TransferOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Config:     f.Config,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "transfer {<number> | <url>} <destination-repo>",
+		Short: "Transfer issue",
+		Args:  cobra.MinimumNArgs(1),
+		Example: heredoc.Doc(`
+					$ gh issue 3252 
+					$ gh issue 3252 cli/cli
+					$ gh issue 3252 --repo cli/cli golang/go
+					$ gh issue 3252 github.com/golang/go
+				`),
```

Comment:
This command is very easy to understand from the `Use` line, so we don't need to provide any examples.